### PR TITLE
Formatting display of data

### PR DESCRIPTION
**Pull request by joshkadis**
Allows specifying a [d3.format](https://github.com/d3/d3-format/blob/master/README.md) string plus some other stuff to change how data is display in yAxis ticks and tooltips. This is pretty basic for now. Will add options for the X axis when applicable, and will turn the text input into a series of select and checkboxes so you can specify, say, percent or currency without looking up the d3 formatting guide every time.

Since NVD3 requires a function for this and the chart data is stored as JSON, there's also a method for converting the JSON object to a `axis.tickFormat` function when rendering a chart embed widget.

### CODE REVIEW

File: app/components/Chart/ChartTypes/NVD3BaseChart/index.js [hunk 1]
```diff
@@ -59,6 +62,21 @@ class BaseChart extends Component {
...
 
+  /**
+   * Create tick formatter function from JSON data
+   *
+   * @param object options Options object
+   * @param string key 'xAxis' or 'yAxis'
+   */
+  _setTickFormat(options, key) {
+    if (options.tickFormatBuilder && options.tickFormatBuilder[key]) {
+      options[key] = update(options[key] || {}, { $merge: { // eslint-disable-line no-param-reassign
```
Comment (nosro) [inline app/components/Chart/ChartTypes/NVD3BaseChart/index.js]: I haven't seen the dollar sign object property prefix before like `$merge` - does that convey special meaning?

File: app/utils/createFormatter.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ *    @param string prepend Any string to prepend
+ *    @param string format D3 format https://github.com/d3/d3-format#locale_format
+ *    @param number multiplier E.g. .001 to transform 1000 into 1
+ *    @param string append  Any string to append
+ * @return function Formatting function
+ */
+export default function(formatData) {
+  return (d) => [
+    formatData.prepend,
+    d3.format(formatData.format)(d * formatData.multiplier),
```
Comment (nosro) [inline app/utils/createFormatter.js]: Any `parseFloat` or `parseInt` bulletproofing needed here to make sure `multiplier` is numeric?

**Comment by nosro** [comment]
😋

File: app/utils/misc.js [hunk 3]
```diff
@@ -25,3 +25,14 @@ export function debounce(func, ...theArgs) {
...
 }
+
+/**
+ * Determine if data is multi-series or single series
+ *
+ * @param array data Data array
+ * @return bool True if multi-series, false if single-series
+ */
+export function dataIsMultiSeries(data) {
+  return (typeof data[0].key !== 'undefined'
```
Comment (jackmarketon) [inline app/utils/misc.js]: `&&` should be at the end of the line, not the beginning... 

I just realized I need to add that to an eslint rule...  so for now it's fine.

**Comment by jackmarketon** [comment]
:octocat:

**Pull request merged**